classification/scripts/models.py

```diff
@@ -0,0 +1,191 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchvision import models
+
+
+# --------------------
+# Utility Modules
+# --------------------
+
+class InitialConv(nn.Module):
+    def __init__(self, in_channels, out_channels):
+        super().__init__()
+        self.conv = nn.Conv2d(in_channels, out_channels, kernel_size=7,
+                              stride=2, padding=3)
+        self.bn = nn.BatchNorm2d(out_channels)
+        self.relu = nn.ReLU(inplace=True)
+
+    def forward(self, x):
+        return self.relu(self.bn(self.conv(x)))
+
+
+class MaskAttention(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, img_feat, mask_feat):
+        if mask_feat.shape[2:] != img_feat.shape[2:]:
+            mask_feat = F.interpolate(mask_feat, size=img_feat.shape[2:],
+                                      mode='bilinear', align_corners=False)
+        attention = torch.sigmoid(mask_feat)
+        return img_feat * attention
+
+
+class ResBlock(nn.Module):
+    def __init__(self, in_channels, out_channels, stride=1):
+        super().__init__()
+        self.conv1 = nn.Conv2d(in_channels, out_channels, kernel_size=3,
+                               stride=stride, padding=1)
+        self.bn1 = nn.BatchNorm2d(out_channels)
+        self.relu = nn.ReLU(inplace=True)
+
+        self.conv2 = nn.Conv2d(out_channels, out_channels, kernel_size=3,
+                               padding=1)
+        self.bn2 = nn.BatchNorm2d(out_channels)
+
+        self.skip = nn.Sequential()
+        if in_channels != out_channels or stride != 1:
+            self.skip = nn.Sequential(
+                nn.Conv2d(in_channels, out_channels, kernel_size=1,
+                          stride=stride),
+                nn.BatchNorm2d(out_channels)
+            )
+
+    def forward(self, x):
+        identity = self.skip(x)
+        out = self.relu(self.bn1(self.conv1(x)))
+        out = self.bn2(self.conv2(out))
+        out += identity
+        return self.relu(out)
+
+
+# --------------------
+# ResNet Siamese Network
+# --------------------
+
+def get_pretrained_resnet34(output_size, freeze=False):
```

Comment:
For code readability I'd have it as a static method in one of the classes or have it in utils.py